client/question/services/code_evaluators/PythonCodeRunnerServiceSpec.js

```diff
@@ -64,6 +64,32 @@ describe('PythonCodeRunnerService', function() {
       $httpBackend.flush();
     });
 
+    it('processes a compile-time syntax error', function() {
+      var code = [
+        'def myFunction(arg):',
+        '    result = arg.rstrip()',
+        '    return result',
+        ''
+      ].join('\n');
+      responseDict.stderr = [
+        'Traceback (most recent call last):',
+        '  File "main.py", line 54, in <module>',
+        '    StudentCode().fmcc, all_tasks_test_inputs[0][0]))',
+        '  File "main.py", line 19, in runTest',
+        '    output = func(input)',
+        '  File "main.py", line 28, in fmcc',
+        '    return 2 / 0',
+        'ZeroDivisionError: integer division or modulo by zero'
+      ].join('\n');
+      $httpBackend.expectPOST('/ajax/compile_code').respond(
+        HTTP_STATUS_CODE_OK, responseDict);
+      spyOn(ServerHandlerService, 'doesServerExist').and.returnValue(true);
+      spyOn(PythonCodeRunnerService,
```

Comment:
Can we not mock this? I understand the idea of keeping a unit test "unit", but this is an internal method called by the compile code function and my original concern when asking for an additional test was to check that the interoperation between the multiple functions works correctly (at least within the scope of compileCodeAsync).

Or we could leave this here and add another test that checks that. Up to you.